.github/actions/upload-flaky-tests/action.yml

```diff
@@ -0,0 +1,44 @@
+name: Report flaky test
+description: Finds flaky tests and creates issues
+
+runs:
+  using: composite
+  steps:
+    - id: flaky-tests
+      name: Find flaky tests
+      if: github.repository == 'stianst/keycloak'
+      shell: bash
+      run: |
+        REPO="${{ github.repository }}"
+        ISSUES_LINK="https://github.com/${{ github.repository }}/issues/"
+        PR="${{ github.event.pull_request.number }}"
+        RUN="https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+        for dir in $(find -type d -name surefire-reports); do
+          for i in $(ls $dir/*.txt); do
+            if ( grep '<<< FAILURE!' $i &>/dev/null ); then
+              TEST="$(echo $i | sed 's/.txt//g' | sed 's/.*\.//g')"
+              LOG="$(cat $i)"
+              
+              TITLE="Flaky test: $TEST"
+              BODY="$RUN"$'\n''```'$'\n'"$LOG"$'\n''```'
+              ISSUE="$(gh issue list --search "$TITLE in:title" --json number --jq .[].number)"
+        
+              if [ "$ISSUE" == "" ]; then
+                LINK="$(gh issue create -t "$TITLE" -b "$BODY" -l "kind/bug,area/ci,flaky-test" | grep $ISSUES_LINK)"
+                ISSUE="$(echo $LINK | sed "s|$ISSUES_LINK||")"
+              else
+                LINK="$(gh issue comment "$ISSUE" -b "$BODY")"
+              fi    
```

Comment:
Actually, yeah it could be an enhancement to check if the flaky test is changed/added by the PR. I would prefer to do that as a follow-up though, as I'd like to get this in for now to be able to experiment with it.

Reply:
I agree with a follow-up. Not sure if a GHT from a PR can actually update an issue (would be a security problem IMHO) if that would be possible.